modules/unsupported/flatgeobuf/src/test/java/org/geotools/data/flatgeobuf/FlatGeobufDataStoreTest.java

```diff
@@ -870,6 +863,36 @@ public void readCountriesBbox() throws IOException {
         }
     }
 
+    @Test
+    public void readCountriesWithNullIntersection() throws IOException {
+        SimpleFeatureSource featureSource = getFeatureSource("countries");
+        SimpleFeatureType schema = featureSource.getSchema();
+        FilterFactory2 ff = CommonFactoryFinder.getFilterFactory2();
+        String geometryPropertyName = schema.getGeometryDescriptor().getLocalName();
+        Filter filter = ff.intersects(ff.property(geometryPropertyName), ff.literal(null));
+        Query query = new Query(schema.getTypeName(), filter);
+        SimpleFeatureCollection featureCollection = featureSource.getFeatures(query);
+        try (SimpleFeatureIterator it = featureCollection.features()) {
+            int count = 0;
+            while (it.hasNext()) {
+                it.next();
+                count++;
+            }
+            assertEquals(0, count);
+        }
```

Comment:
Simpler way, does the same:

```suggestion
        assertEquals(0, DataUtilities.count(features));
```